File: src/api/models/proposal.ts
import { WAXCurrency } from '@/api/models/common.ts';
import { BallotOptions } from '@/api/models/voting.ts';
import { ProposalStatusKey } from '@/constants.ts';

export interface Proposal {
  proposal_id: number;
  proposer: string;
  category: number;
  status: ProposalStatusKey;
  ballot_name: string;
  title: string;
  description: string;
  image_url: string;
  estimated_time: number;
  total_requested_funds: WAXCurrency;
  to_be_paid_funds: WAXCurrency;
  total_paid_funds: WAXCurrency;
  remaining_funds: WAXCurrency;
  deliverables: number;
  deliverables_completed: number;
  reviewer: string;
  ballot_results: BallotOptions[];
  update_ts: string;
  vote_end_time: string;
  road_map: string;
  categories?: Record<string, string>;
}

export interface ProposalContent {
  content: string;
  proposal_id: number;
}

export interface ProposalComment {
  status_comment: string;
  proposal_id: number;
}

export interface DeliverableComment {
  status_comment: string;
  deliverable_id: number;
}

export interface EditProposalRequest {
  proposal_id: number;
  title: string;
  category: string;
  description: string;
  image_url: string;
  estimated_time: number;
  content: string;
  road_map: string;
}

export interface NewProposalRequest extends Omit<EditProposalRequest, 'proposal_id'> {
  deliverables: number;
}

File: notifications/src/notifications/wax_interface/schemas/proposal.py
import typing

from pydantic import BaseModel

from notifications.wax_interface.schemas.types import BallotOptions, ProposalStatus


class Proposal(BaseModel):
    proposal_id: int
    proposer: str
    category: int
    status: ProposalStatus
    ballot_name: str
    title: str
    description: str
    image_url: str
    estimated_time: int
    total_requested_funds: str
    to_be_paid_funds: str
    total_paid_funds: str
    remaining_funds: str
    deliverables: int
    deliverables_completed: int
    reviewer: str
    ballot_results: list[BallotOptions]
    update_ts: str
    vote_end_time: str
    road_map: str
    categories: dict[str, typing.Any] | None = None
